FreeRTOS/FreeRTOS-Kernel
# Fix portSWITCH_TO_USER_MODE() on Armv7-M MPU ports

Description
-----------
A task's privilege level is stored in `ulTaskFlag` member in the TCB. Current implementation of `portSWITCH_TO_USER_MODE()` does not update this flag but just lowers the processor's privilege level. This results in many APIs incorrectly determining task's privilege level and access permissions - 
* xPortIsAuthorizedToAccessBuffer
* xPortIsTaskPrivileged
* xPortIsAuthorizedToAccessKernelObject

This PR fixes the `portSWITCH_TO_USER_MODE()` implementation to correctly update the `ulTaskFlag` member in the TCB before lowering the processor's privilege level.

Test Steps
-----------
Tested on Cortex-M7.

Checklist:
----------



Related Issue
-----------
NA.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
